packages/cube/README.md

```diff
@@ -6,80 +6,116 @@ The `manifest.ttl` file contains a full list of all operations included in this
 ## Installation
 
 ```
-npm i barnard59-cube
+npm install barnard59-cube
 ```
 
 ## Operations
 
-### `cube/buildCubeShape`
+### `toObservation`
 
-TBD
+A pipeline step can use `toObservation` to create
+cube observations according to the [cube.link](https://cube.link/) specification. 
 
-### `cube/toObservation`
+Each input chunk is expected to be a dataset (or quads array)
+with data for a single observation (usually obtained from a CSVW mapping).
+The operation cleans up the data, adding suitable defaults and identifiers.
+
+
+### `buildCubeShape`
+
+The `buildCubeShape` operation derives a SHACL shape from observations.
+
+Each input chunk is expected to be a dataset (or quads array)
+representing a single observation (usually obtained with `toObservation`).
+The operation collects information about the values of observation properties and propagates the chunk.
+Once the input stream is over, it creates and emits a shape describing the properties of the observations.
 
-TBD
 
 
 ## Commands
 
-Peer dependencies must be explicitly installed
+The package provides pipelines to retrieve and validate cube observations and metadata.
 
+Once `barnard59-cube` is installed, run `barnard59 cube` to list the pipelines available as CLI commands.
+You can get help on each one, for example, `barnard59 cube fetch-cube --help`
+
+
+
+
+
+### fetch cube
+
+Pipeline `fetch-cube` queries a given SPARQL endpoint to retrieve all the data and metadata of a given cube.
+
+```bash
+barnard59 cube fetch-cube \
+    --cube https://agriculture.ld.admin.ch/agroscope/PRIFm8t15/2 \
+    --endpoint https://int.lindas.admin.ch/query \
 ```
-npm i barnard59-base barnard59-formats barnard59-http barnard59-rdf barnard59-sparql barnard59-shacl
-```
+For big cubes, it is advisable to use the other fetch commands to separately get the cube parts (for example `fetch-metadata` and `fetch-observations`).
 
-## Cube validation
+### fetch constraint
 
-The following pipelines retrieve and validate cube observations and metadata.
+Pipeline `fetch-constraint` queries a given SPARQL endpoint to retrieve 
+a [concise bounded description](https://docs.stardog.com/query-stardog/#describe-queries) of the constraint of a given cube.
```

Comment:
I don't think it's really important that we're fetching CBD.

Also, it is a Stardog-specific detail we need to remove in the future with a more targeted `CONSTRUCT` query

```suggestion
Pipeline `fetch-constraint` queries a given SPARQL endpoint to retrieve the [constraint shape](https://cube.link/#constraints) of a given cube.
```